commands/url_test.go

```diff
@@ -0,0 +1,68 @@
+package commands
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/cristianoliveira/ergo/proxy"
+)
+
+func TestURL(t *testing.T) {
+
+	tmpfile, err := ioutil.TempFile("", "testaddservice")
+	if err != nil {
+		t.Fatalf("Error creating tempfile: %s", err.Error())
+	}
+
+	defer os.Remove(tmpfile.Name())
+
+	if _, err = tmpfile.Write([]byte("test.dev localhost:9999")); err != nil {
+		t.Fatalf("Error writing to temporary file: %s", err.Error())
+	}
+
+	if err = tmpfile.Close(); err != nil {
+		t.Fatalf("Error closing temp file: %s", err.Error())
+	}
+
+	if err != nil {
+		t.Fatalf("No error expected while initializing config file. Got %s.", err.Error())
+	}
+	config := proxy.Config{}
+	config.ConfigFile = tmpfile.Name()
+	config.Services, err = proxy.LoadServices(config.ConfigFile)
+
+	if err != nil {
+		t.Fatalf("No error expected while loading services from config file. Got %s.", err.Error())
+	}
+
+	service := proxy.Service{}
+	service.Name = config.Services[0].Name
+
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	outC := make(chan string)
+	var buf bytes.Buffer
+
+	go func() {
+		io.Copy(&buf, r)
+		outC <- buf.String()
+	}()
+
+	URL("test.dev", &config)
+
+	w.Close()
+
+	os.Stdout = old
+
+	out := <-outC
```

Comment:
I was thinking, in order to avoid all this setup we could just return a `String` with the current output or send our `Writer` as dependency injection and use `fmt.Fprintl(ourwriter, ...)` in order to get those outputs.

Reply:
Yes. We could.